### PR TITLE
Reconcile terraform plugin framework resources without CLI

### PR DESCRIPTION
**Pull request by erhancagirici**
### Description of your changes

Reconciles terraform plugin framework resources without CLI using the new `TerraformPluginFrameworkExternalClient` no-fork architecture.

I have:

- [ ] Run `make reviewable test` to ensure this PR is ready for review.

### CODE REVIEW

**Comment by erhancagirici** [comment]
/test-examples="examples/ec2/securitygroupingressrule.yaml"

**Comment by erhancagirici** [comment]
/test-examples="examples/cognitoidp/userpoolclient.yaml"

**Comment by erhancagirici** [comment]
/test-examples="examples/simpledb/domain.yaml"

**Comment by erhancagirici** [comment]
/test-examples="examples/appconfig/environment.yaml"

**Comment by erhancagirici** [comment]
/test-examples="examples/appconfig/environment.yaml"

**Comment by erhancagirici** [comment]
/test-examples="examples/ec2/securitygroupingressrule.yaml"

**Comment by erhancagirici** [comment]
/test-examples="examples/cognitoidp/userpoolclient.yaml"

**Comment by erhancagirici** [comment]
/test-examples="examples/simpledb/domain.yaml"

**Comment by erhancagirici** [comment]
/test-examples="examples/appconfig/environment.yaml"

**Comment by erhancagirici** [comment]
/test-examples="examples/ec2/v1beta1/securitygroupingressrule.yaml"

**Comment by erhancagirici** [comment]
/test-examples="examples/simpledb/v1beta1/domain.yaml"

**Comment by erhancagirici** [comment]
/test-examples="examples/appconfig/v1beta1/environment.yaml"

**Comment by erhancagirici** [comment]
/test-examples="examples/ec2/v1beta1/securitygroupegressrule.yaml"

**Comment by erhancagirici** [comment]
/test-examples="examples/ec2/v1beta1/securitygroupingressrule.yaml"

**Comment by erhancagirici** [comment]
/test-examples="examples/cognitoidp/v1beta1/userpoolclient.yaml"

**Comment by erhancagirici** [comment]
/test-examples="examples/simpledb/v1beta1/domain.yaml"

**Comment by erhancagirici** [comment]
/test-examples="examples/appconfig/v1beta1/environment.yaml"

**Comment by erhancagirici** [comment]
/test-examples="examples/elasticache/v1beta1/replicationgroup.yaml"

**Comment by ulucinar** [comment]
/test-examples="examples/cognitoidp/v1beta1/userpoolclient.yaml"

**Comment by ulucinar** [comment]
/test-examples="examples/appconfig/v1beta1/environment.yaml"

**Comment by ulucinar** [comment]
/test-examples="examples/elasticache/v1beta1/replicationgroup.yaml"

**Comment by ulucinar** [comment]
/test-examples="examples/elasticache/v1beta1/replicationgroup.yaml"

**Comment by ulucinar** [comment]
/test-examples="examples/appconfig/v1beta1/environment.yaml"

**Comment by ulucinar** [comment]
/test-examples="examples/simpledb/v1beta1/domain.yaml"

**Comment by ulucinar** [comment]
/test-examples="examples/cognitoidp/v1beta1/userpoolclient.yaml"

**Comment by ulucinar** [comment]
/test-examples="examples/ec2/v1beta1/securitygroupingressrule.yaml"

**Comment by ulucinar** [comment]
/test-examples="examples/ec2/v1beta1/securitygroupegressrule.yaml"

File: config/externalname.go [hunk 1]
```diff
@@ -2651,22 +2680,7 @@ var NoForkExternalNameConfigs = map[string]config.ExternalName{
...
-	"aws_cognito_user_pool_client": cognitoUserPoolClient(),
-	// simpledb
-	//
-	// SimpleDB Domains can be imported using the name
-	"aws_simpledb_domain": config.NameAsIdentifier,
-	// AppConfig Environments can be imported by using the environment ID and application ID separated by a colon (:)
-	// terraform-plugin-framework
-	"aws_appconfig_environment": config.IdentifierFromProvider,
-}
+var CLIReconciledExternalNameConfigs = map[string]config.ExternalName{}
```
Comment (ulucinar) [inline config/externalname.go:2683]: nit: Let's remove this map as it's currently not needed. If we need in the future, then we can add.

File: config/externalname.go [hunk 2]
```diff
@@ -3011,19 +3043,22 @@ func TemplatedStringAsIdentifierWithNoName(tmpl string) config.ExternalName {
...
+		// If an external name is configured for multiple architectures,
+		// Terraform Plugin Framework takes precedence over Terraform
+		// Plugin SDKv2, which takes precedence over CLI architecture.
+		e, configured := TerraformPluginFrameworkExternalNameConfigs[r.Name]
 		if !configured {
-			e, configured = CLIReconciledExternalNameConfigs[r.Name]
+			e, configured = NoForkExternalNameConfigs[r.Name]
+			if !configured {
+				e, configured = CLIReconciledExternalNameConfigs[r.Name]
+			}
```
Comment (ulucinar) [inline config/externalname.go:3061]: Please see the comment above regarding the CLI-based reconciliation map.

File: config/externalname.go [hunk 3]
```diff
@@ -15,9 +15,38 @@ import (
...
+	// simpledb
+	//
+	// SimpleDB Domains can be imported using the name
+	"aws_simpledb_domain": config.NameAsIdentifier,
+
+	// appconfig
+	//
+	// AppConfig Environments can be imported by using the environment ID and application ID separated by a colon (:)
+	// terraform-plugin-framework
+	"aws_appconfig_environment": appConfigEnvironment(),
```
Comment (ulucinar) [inline config/externalname.go:44]: If I'm not mistaken, the external-name syntax is not changing for the resource as we move from the CLI-based implementation to the native fw reconciliation. We are allowed to break APIs with the next release, and if we are breaking APIs, it's important for us to capture this in the release notes.

**Review by ulucinar — APPROVED** [review]
Thanks @erhancagirici, lgtm. We may consider cleaning up the commit history.

File: config/externalname.go (hunk 1 above)
Comment (erhancagirici) [reply config/externalname.go:2683]: lets remove this at CLI removal PR.

**Comment by erhancagirici** [comment]
/test-examples="examples/ec2/v1beta1/securitygroupingressrule.yaml"

**Pull request merged**